client/src/redux/reducers/main.ts

```diff
@@ -4,10 +4,17 @@ import * as Action from '../actionTypes';
 import { MainState, MainAction, SESSION_READ } from '../types';
 
 const initialState: MainState = {
-    isAppLoading: true,
+    isAppLoading: false,
     uuid: '',
     sessionMode: SESSION_READ,
-    cookieConsent: false,
+    projects: [],
+    datasets: [],
```

Comment:
This is a good point. There are a few reasons I did it this way but I am open to alternatives since it's not entirely obvious what the access patterns will be like yet.
1. When viewing a dataset we don't really care which project it's part of so I don't want to have to index through a project to access a dataset
2. Projects are only really an organising and permissions mechanism for datasets. You can well imagine having a side-by-side view of 2 datasets from 2 different projects (so having indexing through projects becomes potentially awkward).
3. I suspect that indexing through projects will turn out to just be noise. We will rarely have to care which project a dataset is in (in the client) so requiring that we access a dataset through it's project, I suspect, will just add unnecessary typing.

Reply:
Makes sense to keep it like this then (for now). I was assuming that you could only have datasets of 1 project open at the same time, but this restriction doesn't really make any sense now that I think about it.